src/middleware/__tests__/apiKeyChecker.unit.test.ts

```diff
@@ -0,0 +1,47 @@
+import fs from 'fs';
+import { AuthenticationError } from '../../errors';
+import { checkApiKey } from '../apiKeyChecker';
+import type { NextFunction, Request, Response } from 'express';
+
+describe('Authorization middleware', () => {
+  let mockRequest: Partial<Request> = {};
+  const mockResponse: Partial<Response> = {};
+  const nextFunction: NextFunction = jest.fn();
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('without x-api-key header', async () => {
+    const expectedResponse = new AuthenticationError('API key not provided in the header "x-api-key"');
```

Comment:
This is a minor point and please feel free to ignore it -- defining the expected response up here removes it a little from when the value is actually used / checked.  It isn't reused, so doing it like this doesn't save us lines of code, for instance.  Maybe worth just having the expected response defined inline on line 23 (this is how our other tests are written for what it's worth).

At the very least I think that we should follow the pattern of:

1. Arrange
2. Act
3. Assert

Described [here](https://automationpanda.com/2020/07/07/arrange-act-assert-a-pattern-for-writing-good-tests/)

Specifically we would have `Arrange` be about defining the values needed to act (not to assert).

TL;DR let's either move this definition down to be part of the "assert" block of the tests, OR (ideally IMO) just define them inline as part of the assert.